### PR TITLE
Add utilities needed by Dolt import path

### PR DESCRIPTION
**Pull request by VinaiRachakonda**

### CODE REVIEW

File: sql/plan/channel_row_source.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+// ChannelRowSource is a sql.Node that wraps a channel as a sql.RowIter.
+type ChannelRowSource struct {
```
Comment (andrew-wm-arthur) [inline sql/plan/channel_row_source.go:25]: this really belongs on the Dolt side if it's not being used in the engine

File: sql/plan/error_wrapper.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+// ErrorHandler wraps a row iter and calls a handler function on each error.
+type ErrorHandler struct {
+	UnaryNode
+	ErrorHandlerFunc
+}
+
+var _ sql.Node = (*ErrorHandler)(nil)
+
+// NewErrorHandlerNode returns a newly created ErrorHandler node.
+func NewErrorHandlerNode(child sql.Node, errorHandler ErrorHandlerFunc) *ErrorHandler {
```
Comment (andrew-wm-arthur) [inline sql/plan/error_wrapper.go:35]: same for this node

**Review by andrew-wm-arthur — APPROVED** [review]
lgtm, a few comments

**Review by zachmu — APPROVED** [review]
LG with caveats noted by andy

**Pull request merged**